language/stdlib/modules/validator_set.mvir

```diff
@@ -32,6 +32,42 @@ module ValidatorSet {
         return;
     }
 
+    // Return the size of the current validator set
+    public size(): u64 acquires T {
+        let validator_set_ref: &Self.T;
+
+        validator_set_ref = borrow_global<T>(0x1D8);
+        return Vector.length<Self.ValidatorInfo>(&move(validator_set_ref).validators);
+    }
+
+    // Return true if addr is a current validator
+    public is_validator(addr: address): bool acquires T {
+        let size: u64;
+        let i: u64;
+        let vector_ref: &Vector.T<Self.ValidatorInfo>;
+        let info_ref: &Self.ValidatorInfo;
+
+        vector_ref = &borrow_global<T>(0x1D8).validators;
+        size = Vector.length<Self.ValidatorInfo>(copy(vector_ref));
+        if (copy(size) == 0) {
+            return false;
+        }
+
+        i = 0;
+        info_ref = Vector.borrow<Self.ValidatorInfo>(copy(vector_ref), copy(i));
+        while (copy(i) < copy(size)) {
+            if (*&copy(info_ref).addr == copy(addr)) {
+            // TODO: below doesn't work, looks like a compiler bug
```

Comment:
Which part doesn't work? The parser or the ir_to_bytecode compiler? Curious to see what was going on.

Reply:
If you patch this PR, uncomment, and try to build the genesis tx, you'll see what's going on. It complains about an invalid type parameter. I suspect it's due to a parse error, but haven't investigated yet.